+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+
+                        lambda_lambda = obj.w.lambda_lambda(ii,jj);
+                        lambda_c = obj.w.lambda_c(ii,jj);
+                        B_max = obj.w.B_max(ii,jj);
+                        pi_lambda = obj.w.pi_lambda(ii,jj);
+                        pi_c = obj.w.pi_c(ii,jj);
+                        eta = obj.w.eta(ii,jj);
+                        nu = obj.w.nu(ii,jj);
+
+                        obj.g.pi_lambda_or(ii,jj) = {[pi_lambda-sigma_lambda_F;pi_lambda-sigma_lambda_B;sigma_lambda_F+sigma_lambda_B-pi_lambda],0,inf};
+                        obj.g.pi_c_or(ii,jj) = {[pi_c-sigma_c_F;pi_c-sigma_c_B;sigma_c_F+sigma_c_B-pi_c],0,inf};
+
+                        % kkt conditions for min B, B>=sigmaB, B>=sigmaF
+                        kkt_max = [1-lambda_c-lambda_lambda;
+                            B_max-pi_lambda;
+                            B_max-pi_c];
+                        obj.g.kkt_max(ii,jj) = {kkt_max,
+                            [0*ones(dims.n_lambda,1);0*ones(dims.n_lambda,1);0*ones(dims.n_lambda,1)],
+                            [0*ones(dims.n_lambda,1);inf*ones(dims.n_lambda,1);inf*ones(dims.n_lambda,1)]};
+
+                        obj.G.step_eq_kkt_max(ii,jj) = {[(B_max-pi_lambda);(B_max-pi_c)]};
+                        obj.H.step_eq_kkt_max(ii,jj) = {[lambda_lambda;lambda_c]};
+                        
+                        % eta calculation
+                        eta_const = [eta-pi_c;eta-pi_lambda;eta-pi_c-pi_lambda+B_max];
+                        obj.g.eta_const(ii,jj) = {eta_const,
+                            [-inf*ones(dims.n_lambda,1);-inf*ones(dims.n_lambda,1);zeros(dims.n_lambda,1)],
+                            [zeros(dims.n_lambda,1);zeros(dims.n_lambda,1);inf*ones(dims.n_lambda,1)]};
+
+                        obj.g.nu_or(ii,jj) = {[nu-eta;sum(eta)-nu],0,inf};
+
+                        % the actual step eq conditions
+                        M=obj.p.T()/opts.N_stages;
+                        delta_h = obj.w.h(ii,jj) - obj.w.h(ii,jj-1);
+                        step_equilibration = [delta_h + (1/h0)*nu*M;
+                            delta_h - (1/h0)*nu*M];
+                        obj.g.step_equilibration(ii,jj) = {step_equilibration,[0;-inf],[inf;0]};
+                    end
+                end
+            end
+        end
+
+        function populate_problem(obj)
+            obj.create_variables();
+            obj.generate_direct_transcription_constraints();
+            obj.generate_complementarity_constraints();
+            obj.generate_step_equilibration_constraints();
+
+            obj.populated = true;
+        end
+
+        function create_solver(obj, solver_options, plugin)
+            if ~obj.populated
+                obj.populate_problem();
+            end
+
+            if ~exist('plugin')
+                plugin = 'scholtes_ineq';
+            end
+
+            % Sort by indices to recover almost block-band structure.
+            obj.w.sort_by_index();
```

Comment:
not related to this PR, but plotting the Jacobians of the constrains with  FESD ON and OFF would be interesting (also after introducing a clock state for the integration, to get rid of sum h constraints that interfere with a block band structure, cf. https://github.com/nurkanovic/nosnoc/issues/102)

Reply:
Agreed. Added a TODO, that perhaps we also want to do some smarter re-ordering for a better structure.